lib/ts/ingredients/smsdelivery/services/twilio.ts

```diff
@@ -0,0 +1,110 @@
+/* Copyright (c) 2021, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+import OverrideableBuilder from "supertokens-js-override";
+import * as Twilio from "twilio";
+
+export interface TwilioServiceConfig {
+    accountSid: string;
+    authToken: string;
+    /**
+     * only one of "from" and "messagingServiceSid" should be passed.
+     * if both are passed, we should throw error to the user
+     * saying that only one of them should be set. this is because
+     * both parameters can't be passed while calling twilio API.
+     * if none of "from" and "messagingServiceSid" is passed, error
+     * should be thrown.
+     */
+    from?: string;
+    messagingServiceSid?: string;
+    opts?: Twilio.Twilio.TwilioClientOptions;
+}
+
+export type TypeNormalisedInput<T> = {
+    twilioSettings: {
+        accountSid: string;
+        authToken: string;
+        opts?: Twilio.Twilio.TwilioClientOptions;
+    } & (
+        | {
+              from: string;
+              messagingServiceSid: undefined;
+          }
+        | {
+              messagingServiceSid: string;
+              from: undefined;
+          }
```

Comment:
```suggestion
        | {
              from: string;
          }
        | {
              messagingServiceSid: string;
          }
```

Reply:
If you accept my change for changes in `TwilioServiceConfig`, you won't need `TypeNormalisedInput` anymore.

Reply:
removed